molgenis-app/src/main/java/org/molgenis/app/promise/mapper/RadboudSampleMap.java

```diff
@@ -0,0 +1,433 @@
+		{
+			materialTypeIds.add("WHOLE_BLOOD");
+		}
+
+		if ("1".equals(radboudSampleEntity.getString(XML_BLOEDPLASMA)))
+		{
+			materialTypeIds.add("PLASMA");
+		}
+
+		if ("1".equals(radboudSampleEntity.getString(XML_BLOEDSERUM)))
+		{
+			materialTypeIds.add("SERUM");
+		}
+
+		if ("1".equals(radboudSampleEntity.getString(XML_WEEFSELSOORT)))
+		{
+			materialTypeIds.add("TISSUE_PARAFFIN_EMBEDDED");
+		}
+		else if ("2".equals(radboudSampleEntity.getString(XML_WEEFSELSOORT)))
+		{
+			materialTypeIds.add("TISSUE_FROZEN");
+		}
+
+		if ("1".equals(radboudSampleEntity.getString(XML_URINE)))
+		{
+			materialTypeIds.add("URINE");
+		}
+
+		if ("1".equals(radboudSampleEntity.getString(XML_SPEEKSEL)))
+		{
+			materialTypeIds.add("SALIVA");
+		}
+
+		if ("1".equals(radboudSampleEntity.getString(XML_FECES)))
+		{
+			materialTypeIds.add("FECES");
+		}
+
+		if ("1".equals(radboudSampleEntity.getString(XML_RNA)) || "1"
+				.equals(radboudSampleEntity.getString(XML_RNABEENMERG)))
+		{
+			materialTypeIds.add("MICRO_RNA");
+		}
+
+		if ("1".equals(radboudSampleEntity.getString(XML_GASTROINTMUC)) || "1"
+				.equals(radboudSampleEntity.getString(XML_LIQUOR)) || "1"
+				.equals(radboudSampleEntity.getString(XML_CELLBEENMERG)) || "1"
+				.equals(radboudSampleEntity.getString(XML_MONONUCLBLOED)) || "1"
+				.equals(radboudSampleEntity.getString(XML_MONONUCMERG)) || "1"
+				.equals(radboudSampleEntity.getString(XML_GRANULOCYTMERG)) || "1"
+				.equals(radboudSampleEntity.getString(XML_MONOCYTMERG)) || "1"
+				.equals(radboudSampleEntity.getString(XML_MICROBIOOM)))
+		{
+			materialTypeIds.add("OTHER");
+		}
+
+		return materialTypeIds;
+	}
+
+	private Set<String> collectOmicsIds(Entity radboudSampleEntity)
+	{
+		Set<String> omicsTypeIds = new HashSet<>();
+
+		if ("1".equals(radboudSampleEntity.getString(XML_GWASOMNI)) || "1"
+				.equals(radboudSampleEntity.getString(XML_GWAS370CNV)) || "1"
+				.equals(radboudSampleEntity.getString(XML_EXOOMCHIP)))
+		{
+			omicsTypeIds.add("GENOMICS");
+		}
+
+		return omicsTypeIds;
+	}
+
+	private Set<String> collectSexIds(Entity radboudSampleEntity)
+	{
+		Set<String> genderTypeIds = new HashSet<>();
+
+		String genderValue = radboudSampleEntity.getString(XML_GENDER);
+		if ("1".equals(genderValue))
+		{
+			genderTypeIds.add("FEMALE");
+		}
+		if ("2".equals(genderValue))
+		{
+			genderTypeIds.add("MALE");
+		}
+		if ("3".equals(genderValue))
+		{
+			genderTypeIds.add("UNKNOWN");
+		}
+
+		return genderTypeIds;
+	}
+
+	private Integer collectAge(Entity radboudSampleEntity)
+	{
+		String birthDate = radboudSampleEntity.getString(XML_BIRTHDATE);
+		if (birthDate != null && !birthDate.isEmpty())
+		{
+			LocalDate start = parse(birthDate, ISO_DATE_TIME);
+			LocalDate end = now();
+			Long age = YEARS.between(start, end);
+			return age.intValue();
+		}
+		else
+		{
+			return null;
+		}
+	}
+
+	private class AggregatedSampleInfo
+	{
+		private Set<String> materialIds = newHashSet();
```

Comment:
This means the order of the Ids will be completely random, and probably confusing.
A LinkedHashSet would keep the order in which they were collected.
A TreeSet would sort alphabetically (probably the nicest?)